src/pages/Member/SignUp/components/EmailSignUp.js

```diff
@@ -0,0 +1,204 @@
+import React, { useState } from 'react';
+import styled, { css } from 'styled-components';
+import { Link } from 'react-router-dom';
+import Input from '../../components/Input';
+import Button from '../../components/Button';
+import SignUpOrSignIn from '../../components/SignUpOrSignIn';
+import OrDivider from '../../components/OrDivider';
+import { CHECKBOXES } from '../../data/checkBoxes';
+import { ReactComponent as LogoKakao } from '../../../../assets/LogoKakao.svg';
+
+function EmailSignUp({ title }) {
+  const [isChecked, setIsChecked] = useState([false, false, false, false]);
+
+  const handleSingleCheck = event => {
+    const { checked, id } = event.target;
+    let statusUpdatedArray = [...isChecked];
+
+    if (checked) {
+      statusUpdatedArray.splice(id - 1, 1, true);
+      setIsChecked(statusUpdatedArray);
+    } else {
+      statusUpdatedArray.splice(id - 1, 1, false);
+      setIsChecked(statusUpdatedArray);
+    }
+  };
```

Comment:
클릭한 박스의 이전 값`checked`을 알 필요는 없고, `!`연산자로 이전 값을 반대로 바꾸어 주기만 하면 될 것 같아요.
또한 statusUpdatedArray를 `let`으로 말고 `const`로 선언해주세요.
```suggestion
  const handleSingleCheck = event => {
    const { id } = event.target;
    const statusUpdatedArray = [...isChecked];

    // 로직
  };
```

Reply:
감사합니다! 리팩토링 기간에 이 부분 꼭 반영해보겠습니다!

Reply:
@JHyeon0 

장현님! 이 부분 코드를 리팩토링 해보았습니다!
동작상 문제가 없는 것 같아 바로 merge를 시켰지만요...💦

장현님께서 생각하신 이미지가 맞으시길 바랍니다...

https://github.com/wecode-bootcamp-korea/fullstack1-2nd-SKYROCKET-frontend/pull/19